query-builder/src/main/java/com/datastax/oss/driver/api/querybuilder/QueryBuilderDsl.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright DataStax, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.datastax.oss.driver.api.querybuilder;
+
+import com.datastax.oss.driver.api.core.CqlIdentifier;
+import com.datastax.oss.driver.api.querybuilder.select.SelectFrom;
+import com.datastax.oss.driver.internal.querybuilder.select.DefaultSelect;
+
+/** A Domain-Specific Language to build CQL queries using Java code. */
+public interface QueryBuilderDsl {
```

Comment:
I added `Dsl` to avoid a name clash in an earlier prototype, but it's not necessary anymore, so I'm fine going back to `QueryBuilder` if we prefer keeping the same name as 3.x.